source/extensions/transport_sockets/alts/tsi_frame_protector.h

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include "envoy/buffer/buffer.h"
+
+#include "extensions/transport_sockets/alts/grpc_tsi.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace TransportSockets {
+namespace Alts {
+
+/**
+ * A C++ wrapper for tsi_frame_protector interface.
+ * For detail of tsi_frame_protector, see
+ * https://github.com/grpc/grpc/blob/v1.10.0/src/core/tsi/transport_security_interface.h#L70
+ *
+ * TODO(lizan): migrate to tsi_zero_copy_grpc_protector for further optimization
+ */
+class TsiFrameProtector final {
+public:
+  explicit TsiFrameProtector(CFrameProtectorPtr&& frame_protector);
+
+  /**
+   * Wrapper for tsi_frame_protector_protect
+   * @param input supplies the input buffer, the method will drain it when it is processed.
```

Comment:
input supplies the input buffer -> input supplies the data to protect.


ditto below for unprotect